src/vehicle.cpp

```diff
@@ -5182,11 +5182,14 @@ bool vehicle::fire_turret (int p, bool /* burst */ )
         if( liquid_fuel < 1 || charges < 1 ) {
             return false;
         }
-        itype *am_type = item::find_type( amt );
+        itype *am_type = item::find_type( default_ammo( amt ) );
+        itype *fake;
+        bool fake_ammo = false;
         if( !am_type->ammo ) {
-            // Let non-ammo-typed ammo be used too
-            // 0 in all stats, depends on gun to provide everything
-            am_type = item::find_type( "fake_ammo" );
+            // Alter ammo type temporarily
+            fake = item::find_type( "fake_ammo" );
+            std::swap( am_type->ammo, fake->ammo );
```

Comment:
```
"type": "COMESTIBLE",
... existing water properties ...
"ammo_data": {
    "damage": ... specific data for usage as ammo here ...
}
```
OK, this will work.
That's what I meant by dual type item.